objects/src/errors.rs

```diff
@@ -38,11 +38,12 @@ pub enum AccountError {
     MapsUpdateToNonMapsSlot(u8, StorageSlotType),
     NonceNotMonotonicallyIncreasing { current: u64, new: u64 },
     SeedDigestTooFewTrailingZeros { expected: u32, actual: u32 },
-    StorageSlotInvalidValueArity { slot: u8, expected: u8, actual: u8 },
     StorageSlotIsReserved(u8),
-    StorageSlotMapOrArrayNotAllowed(u8, StorageSlotType),
-    StorageMapNotFound(u8),
-    StorageMapTooManyMaps { expected: usize, actual: usize },
+    StorageSlotNotMap(u8),
+    StorageSlotNotValue(u8),
+    StorageIndexOutOfBounds(u8),
+    StorageTooManySlots(u64),
+    StubDataIncorrectLength(usize, usize),
```

Comment:
We might need `StorageSlotIsReserved` for faucets reserved slots when trying to write to it, no?

Reply:
I think because we offset storage automatically, this should never happen. We will need to make sure that for faucets, no procedure has storage offset of `0` - but I think this is covered by https://github.com/0xPolygonMiden/miden-base/issues/864.